engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -2220,8 +2228,28 @@ namespace dmGraphics
         *byte_offset_end_out = byte_offset;
     }
 
-    static void CreateProgram(VulkanContext* context, Program* program, ShaderModule* vertex_module, ShaderModule* fragment_module)
+    static bool CreateProgram(VulkanContext* context, Program* program, ShaderModule* vertex_module, ShaderModule* fragment_module)
     {
+        // Check all fragment inputs if they are available as outputs from the vertex shader
+        for (uint32_t i = 0; i < fragment_module->m_InputCount; ++i)
+        {
+            bool input_found = false;
+            for (uint32_t j = 0; j < vertex_module->m_OutputCount; ++j)
+            {
+                if (fragment_module->m_Inputs[i].m_NameHash == vertex_module->m_Outputs[j].m_NameHash)
+                {
+                    input_found = true;
+                    break;
+                }
+            }
+
+            if (!input_found)
+            {
+                dmLogError("Input of fragment shader '%s' not written by vertex shader", fragment_module->m_Inputs[i].m_Name);
+                return false;
+            }
```

Comment:
I feel like this check should be in the shader build pipeline?

At runtime we don't change the vertex/fragment programs, only the whole material. So this situation shouldn't come up at runtime?